### PR TITLE
Signin button UX fixed when user already loggedin

### LINKED ISSUE #263: Fix Sign In button UX for logged-in users
### Description:
`Sign in with Github` button showing showing up in the Navbar while refreshing/navigating among the Pages, when user is already Loggedin.

https://github.com/Real-Dev-Squad/todo-action-items/assets/69259490/b20cfafa-b094-4068-8492-8d3117dbc57b

### PR DESCRIPTION
**Pull request by kotesh-arya**
### Issue
 Closes #263 

### What is the change?

- Added a **.hidden** class to the `Sign in with Github` button, which initially hides it.
- After the SELF-API call being made, based on it's success/fail the button will be shown acccordingly.
- In the error case of the API-call I have targeted the signin-button from the DOM and removed the **.hidden** class on it.
- During the API call I have added Shimmer effect to avoid a Layout shift after the User Name, profile shows up.

### Is Development Tested?

- [x] Yes
- [ ] No

### Before / After Change Screenshots

Before :

https://github.com/Real-Dev-Squad/todo-action-items/assets/69259490/b20cfafa-b094-4068-8492-8d3117dbc57b

After :

Desktop:

https://github.com/Real-Dev-Squad/website-welcome/assets/69259490/2adeedad-c322-41b4-9206-72728e8945e3

Mobile:

https://github.com/Real-Dev-Squad/website-welcome/assets/69259490/677de35b-b8e4-44df-a5e3-0f72f9090e38

### CODE REVIEW

File: js/navbar.js [hunk 1]
```diff
@@ -11,8 +11,6 @@ window.addEventListener('DOMContentLoaded', (event) => {
       : 'c4a84431feaf604e89d1';
   userLoginEl.href = `https://github.com/login/oauth/authorize?client_id=${clientId}&state=${checkURL}`;
 });
-const hamburger = document.querySelector('.hamburger');
-const navMenu = document.querySelector('.nav-menu');
```
Comment (rohan09-raj) [inline js/navbar.js:15]: Why this is removed ? Can we please check the mobile view if it's working fine or not ?
Also if possible you can share the recordings for it

Comment (kotesh-arya) [reply js/navbar.js:15]: The variables **hamburger** & **navMenu** were already on the top that too using `const`. It gave me error when I started working on this issue, so I found them redundant and removed.
Sharing a Screenshot for reference:
image

Comment (kotesh-arya) [reply js/navbar.js:15]: Here's the recording for Mobile view: 

https://github.com/Real-Dev-Squad/website-welcome/assets/69259490/bf6725da-64cf-41a1-9385-e837e76f7ef1

Comment (kotesh-arya) [reply js/navbar.js:15]: The toggling is working fine 🙂

https://github.com/Real-Dev-Squad/website-welcome/assets/69259490/91ffb16e-5f26-4b5a-a039-5230236f171d

**Review by rohan09-raj — APPROVED** [review]

**Pull request merged**